app/models/ticket.rb

```diff
@@ -9,13 +9,28 @@ class Ticket < ApplicationRecord
   has_many :ticket_stakeholders
   belongs_to :metadata, polymorphic: true
 
+  # user_stakeholders will have the list of stakeholders where the user is part of. For example,
+  # if a normal user X requests for a change by creating a ticket, the stakeholders list will be
+  # [X, WRT] (WRT is added as stakeholder because WRT is responsible for taking action on the
+  # ticket). If a WRT member fetches the ticket data, the value of user_stakeholders will be [WRT]
+  # and if the normal user fetches the ticket data, the value of user_stakeholders will be [X]. If
+  # the ticket is created by a WRT member, then the value user_stakeholders will be [X, WRT] because
+  # the user can be any of the two stakeholders.
   def user_stakeholders(user)
     return [] if user.nil?
     ticket_stakeholders.select do |ticket_stakeholder|
       user.active_roles.where(group: ticket_stakeholder.stakeholder).any? || user == ticket_stakeholder.stakeholder
     end
```

Comment:
I think this existing method body may now be able to benefit from your new scopes. I might be reading things wrong here, but isn't this just `ticket_stakeholders.belongs_to_user(user) + ticket_stakeholders.belongs_to_group(user.active_groups)``?

Reply:
Oh yes, I didn't think of that. Thanks for pointing it, done.